extensions/amp-youtube/1.0/component.js

```diff
@@ -115,21 +114,52 @@ export function Youtube({
 
   src = addParamsToUrl(src, params);
 
+  // Passed through to imperative handle below.
+  const infoRef = useRef(
+    dict({
+      'currentTime': 0,
+      'duration': NaN,
+    })
+  );
+
+  useImperativeHandle(
+    ref,
+    () => {
+      const handle = {};
+      for (const key in infoRef.current) {
+        const get = () => infoRef.current[key];
+        Object.defineProperty(handle, key, {get});
+      }
+      return handle;
+    },
+    [infoRef]
+  );
+
   const onMessage = ({data, currentTarget}) => {
-    data = objOrParseJson(data);
-    if (data == null) {
+    const parsedData = objOrParseJson(data);
+    if (!parsedData) {
       return;
     }
-    if (data.event == 'initialDelivery') {
+
+    const {'event': event, 'info': info} = parsedData;
+
+    if (event == 'initialDelivery') {
       dispatchVideoEvent(currentTarget, VideoEvents.LOADEDMETADATA);
       return;
     }
-    const {info} = data;
-    if (info == undefined) {
+
+    if (!info) {
       return;
     }
+
+    for (const key in infoRef.current) {
+      if (info[key] != null) {
+        infoRef.current[key] = info[key];
```

Comment:
Does this create new keys on `infoRef.current`, or is it always `currentTime` and `duration`. Can we add an assert to verify this?

Reply:
Ohhh, I misread and though we were iterating `info`, not `infoRef.current`.